xcape/core.py

```diff
@@ -159,26 +159,26 @@ def _calc_cape_gufunc(p, t, td, ps, ts, tds, *args, **kwargs):
         if kwargs['source']=='most-unstable':
             signature += ",(),()"
             output_dtypes = output_dtypes + ('i4','f4')
-
+        print(output_dtypes)
         return da.apply_gufunc(_calc_cape_numpy, signature,
                                p, t, td, ps, ts, tds, 
-                               output_dtypes=('f4','f4'),
+                               output_dtypes=output_dtypes,#('f4','f4'),
                                axis=-1,
                                vectorize=False,
                                **kwargs)
     
     elif (kwargs['vertical_lev']=='pressure'):
-        print('here')
         pres_lev_pos = args[0]
         signature = "(i),(i),(i),(),(),(),()->(),()"
         output_dtypes = ('f4','f4')
         if kwargs['source']=='most-unstable':
             signature += ",(),()"
             output_dtypes = output_dtypes + ('i4','f4')
 
+        print(output_dtypes)
         return da.apply_gufunc(_calc_cape_numpy, signature,
                                p, t, td, ps, ts, tds, pres_lev_pos,
-                               output_dtypes=('f4','f4'),
+                               output_dtypes=output_dtypes,#('f4','f4'),
```

Comment:
you had a fixed ('f4','f4') instead it should be the varying `output_dtypes`
it didn't fail te tests so I wonder if it understood the type or what?